specifyweb/frontend/js_src/lib/components/QueryBuilder/Export.tsx

```diff
@@ -87,16 +121,21 @@ export function QueryExportButtons({
         <QueryButton
           disabled={fields.length === 0}
           showConfirmation={showConfirmation}
-          onClick={(): void =>
-            doQueryExport(
-              '/stored_query/exportcsv/',
-              userPreferences.get(
-                'queryBuilder',
-                'behavior',
-                'exportFileDelimiter'
-              )
-            )
-          }
+          onClick={(): void => {
+            selectedRows.size === 0
+              ? doQueryExport(
+                  '/stored_query/exportcsv/',
+                  userPreferences.get(
+                    'queryBuilder',
+                    'behavior',
+                    'exportFileDelimiter'
+                  )
+                )
+              : downloadFile(
+                  'querySelectionExport.tsv',
```

Comment:
Can you come up with more helpful name? camelCase name is for programmers, not for regular users 😄 

Take a look at something like this for inspiration: https://github.com/specify/specify7/blob/60dd907699ea82e2aeb42a1eaa69d352e745398e/specifyweb/frontend/js_src/lib/components/HomePage/AboutSpecify.tsx#L48
 be sure to include query name (and handle the case when query doesn't have a name because it's a new query) and include date to keep file name unique

Ideally, you should use the same naming convention as the back-end.
Though, I just noticed that back-end's naming convention is quite bad too.

Thus, afterward, could you also check all the places in the back-end where files are created/downloaded?

<details>
<summary>Hint on how to find them</summary>
In python, to create a file, you would do something like `with open(` - do global search on that.

</details>